SCClassLibrary/Common/Collections/Event.sc

```diff
@@ -44,17 +47,9 @@ Event : Environment {
 		//		^this.delta
 	}
 
-	// this[\isRest] may be nil
 	isRest {
-		^this[\isRest] == true
-		or: { this[\type] == \rest
-			or: {
-				this.use {
-					parent ?? { parent = defaultParentEvent };
-					~detunedFreq.value.isRest
-				}
-			}
-		}
+		_Event_IsRest
+		^false
```

Comment:
To be more specific: if a primitive returns `errNone`, then the top of the stack is the return from the method invoking the primitive. If the primitive returns `errException`, control returns back to the language interpreter as a fallback. The fallback *may* throw an error, but this is not mandatory -- the method may also take action that is impossible or inconvenient in a primitive, and return a successful value. See Symbol:envirPut for an example of this.

```
	envirPut { arg aValue;
		_Symbol_envirPut
		currentEnvironment.put(this, aValue);
		^aValue
	}
```

I do see what you're thinking, but it's based on the assumption that a primitive returns with either success or failure. That's a faulty assumption. There's a third option: let the language finish a job that the primitive couldn't do completely.